classes/class-p4-post-campaign.php

```diff
@@ -315,98 +315,163 @@ private static function campaign_field(
 			register_post_meta( self::POST_TYPE, $meta_key, $args );
 		}
 
+		private static function get_theme_defaults( $theme_json ) {
+			$defaults = [];
+
+			foreach ( $theme_json['fields'] as $field ) {
+				if ( isset( $field['configurations'] ) && isset( $field['configurations']['default'] )) {
+					$defaultConfiguration = $field['configurations']['default'];
+					if ( isset( $field['configurations'][$defaultConfiguration]['default'] ) ) {
+						$defaults[ $field['id'] ] = $field['configurations'][$defaultConfiguration]['default'];
+					}
+				} else if ( isset( $field['default'] )) {
+					$defaults[ $field['id'] ] = $field['default'];
+				}
+			}
+
+			return $defaults;
+		}
+
 		/**
 		 * Determine the css variables for a certain post.
 		 *
 		 * @param array $meta The meta containing the variable values.
 		 * @return array The values that will be used for the css variables.
 		 */
 		public static function css_vars( array $meta ): array {
-			// Set specific CSS for Montserrat.
-			$special_weight_fonts = [
-				'Montserrat'       => '900',
-				'Montserrat_Light' => '500',
-			];
-
-			$campaigns_font_map = [
-				'default'   => 'lora',
-				'antarctic' => 'sanctuary',
-				'arctic'    => 'Save the Arctic',
-				'climate'   => 'Jost',
-				'forest'    => 'Kanit',
-				'oceans'    => 'Montserrat',
-				'oil'       => 'Anton',
-				'plastic'   => 'Montserrat',
-			];
+			$theme = self::get_theme( $meta );
 
-			$theme = empty( $meta['theme'] )
-								? 'default'
-								: $meta['theme'];
+			// TODO: Use wp_safe_remote_get?
+			// TODO: Handle errors
+			$theme_json = json_decode(
+				file_get_contents( __DIR__ . '/../campaign_themes/' . $theme . '.json' ),
+				true
+			);
 
-			$header_primary_font = empty( $meta['campaign_header_primary'] )
-															? $campaigns_font_map[ $theme ]
-															: $meta['campaign_header_primary'];
+			$defaults = self::get_theme_defaults( $theme_json );
 
-			$body_font = empty( $meta['campaign_body_font'] )
-										? $campaigns_font_map[ $theme ]
-										: $meta['campaign_body_font'];
+			// Replace the defaults with the campaign options where applicable
+			// 'array_intersect_key' removes the keys from $meta that are not present in the $defaults.
+			// 'array_filter' removes empty (falsey?) values from the resulting array.
+			$css_vars = array_filter( array_intersect_key( $meta, $defaults ) );
```

Comment:
Not sure how we'll handle the defaults. Will there always be a default value? No? What's the best strategy? 🤔